libs/api/domains/auth/src/lib/resolvers/main.resolver.ts

```diff
@@ -0,0 +1,50 @@
+import { Query, Mutation, Resolver, Args } from '@nestjs/graphql'
+import { UseGuards } from '@nestjs/common'
+
+import { CurrentUser, IdsUserGuard } from '@island.is/auth-nest-tools'
+import type { User } from '@island.is/auth-nest-tools'
+
+import { UpdateDelegationInput, DelegationInput } from '../dto'
+import { Delegation, AuthScope } from '../models'
+import { AuthService } from '../auth.service'
+
+const ignore404 = (e: Error) => {
+  if ((e as any).status !== 404) {
+    throw e
+  }
+}
+
+@UseGuards(IdsUserGuard)
+@Resolver()
+export class MainResolver {
+  constructor(private authService: AuthService) {}
+
+  @Query(() => [Delegation], { name: 'authActorDelegations' })
+  getDelegations(@CurrentUser() user: User): Promise<Delegation[]> {
+    return this.authService.getActorDelegations(user)
+  }
+
```

Comment:
actorDelegations is a completely separate query which we should not call from the access control page. We should create a `delegations` query which returns results for the CurrentUser instead of CurrentActor.

So if you are logged in to a company and open the access control page, you are managing delegations for the company, but the dropdown shows the (actor) delegations for you as a person.

Reply:
Yeah, this is a result of renaming the resolver from `auth.resolver` to `main.resolver`, since resolvers can be generic (thus `main`) or individual type resolvers (e.g. `delegation.resolver`).
So this is not connected to the access control page and IMO doesn't need any further actions.
Please raise your concern if you disagree :)

Reply:
Ok. Just saw that this resolver is named "getDelegations" and you implemented a "getDelegation" resolver, and we also need to get a list of delegation for the access control page.

So these two should be named "getActorDelegations" (authActorDelegations) and "getDelegation" (authDelegation). Then we should create a "getDelegations" (authDelegations) resolver.

Reply:
Agree!